airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/csv_parser.py

```diff
@@ -47,13 +51,95 @@ async def infer_schema(
             doublequote=config_format.double_quote,
             quoting=config_to_quoting.get(config_format.quoting_behavior, csv.QUOTE_MINIMAL),
         )
-        with stream_reader.open_file(file, self.file_read_mode, logger) as fp:
-            # todo: the existing InMemoryFilesSource.open_file() test source doesn't currently require an encoding, but actual
-            #  sources will likely require one. Rather than modify the interface now we can wait until the real use case
+        with stream_reader.open_file(file, file_read_mode, logger) as fp:
             headers = self._get_headers(fp, config_format, dialect_name)
-            schema = {field.strip(): {"type": "string"} for field in headers}
-            csv.unregister_dialect(dialect_name)
-            return schema
+
+            fp.seek(0)
+            # we assume that if we autogenerate columns, it is because we don't have headers
+            # if a user wants to autogenerate_column_names with a CSV having headers, he can skip rows
+            rows_to_skip = (
```

Comment:
I'm not sure this is right - what if the user wants to skip rows before and after the header, but still read the header?